src/printer/scalebar.js

```diff
@@ -0,0 +1,201 @@
+import { getPointResolution, METERS_PER_UNIT } from 'ol/proj';
+import ProjUnits from 'ol/proj/Units';
+import { assert } from 'ol/asserts.js';
+
+export function printScaleBar(ctx, frameState, spec) {
```

Comment:
All the functions of the project are typed with typescript annotation, please do it as well.
If you use an object like `scaleBar` params, create a new type for the options, thanks.

Reply:
... and `ScaleBarParams` for the functions.